[Bug Fix] Resolve invalid model name error for Gemini Imagen models

ikaadil opened the pull request:
<!-- e.g. "Implement user authentication feature" -->

## Relevant issues
Fixes: https://github.com/BerriAI/litellm/issues/13851
Related to: https://github.com/BerriAI/litellm/issues/13846

<!-- e.g. "Fixes #000" -->

## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [ ] I have Added testing in the [`tests/litellm/`](https://github.com/BerriAI/litellm/tree/main/tests/litellm) directory, **Adding at least 1 test is a hard requirement** - [see details](https://docs.litellm.ai/docs/extras/contributing_code)
- [ ] I have added a screenshot of my new test passing locally 
- [x] My PR passes all unit tests on [`make test-unit`](https://docs.litellm.ai/docs/extras/contributing_code)
- [x] My PR's scope is as isolated as possible, it only solves 1 specific problem


## Type

<!-- Select the type of Pull Request -->
<!-- Keep only the necessary ones -->

🆕 New Feature
🐛 Bug Fix
🧹 Refactoring
📖 Documentation
🚄 Infrastructure
✅ Test

## Changes

ikaadil committed:
fix: resolve invalid model name error for Gemini Imagen models (#13851)

- Fix URL construction in Gemini image generation to strip 'gemini/' prefix
- Google AI API expects base model name without the prefix
- Update model references and pricing information for consistency
- Remove outdated image generation pricing entries

Fixes issue where models like 'gemini/imagen-4.0-fast-generate-preview-06-06'
were being rejected by the Google AI API due to incorrect URL formatting.

krrishdholakia commented on model_prices_and_context_window.json:
```diff
@@ -10180,18 +10180,6 @@
         "mode": "image_generation",
         "source": "https://cloud.google.com/vertex-ai/generative-ai/pricing"
     },
-    "vertex_ai/imagen-4.0-generate-preview-06-06": {
```
is the value invalid on vertex ai as well or just google ai studio?

ikaadil replied:
Yes. Invalid on both.

ikaadil committed:
refactor: simplify URL construction for Gemini image generation

- Remove unnecessary model name prefix stripping
- Directly use the model name in the API URL construction

This change streamlines the URL generation process for the Google AI API, ensuring compatibility with model names without the 'gemini/' prefix.

Signed-off-by: Ifta Khairul Alam Adil <ikaadil007@gmail.com>

ikaadil committed:
refactor: remove unnecessary whitespace in GoogleImageGenConfig

- Cleaned up the code by removing an extra blank line in the GoogleImageGenConfig class.
- This minor adjustment improves code readability without affecting functionality.

Signed-off-by: Ifta Khairul Alam Adil <ikaadil007@gmail.com>

ikaadil commented on litellm/llms/gemini/image_generation/transformation.py:
```diff
@@ -88,14 +88,18 @@ def get_complete_url(
             or self.DEFAULT_BASE_URL
         )
 
+        # Strip the "gemini/" prefix from the model name for the API call
```
Done

ikaadil committed:
Merge branch 'main' into fix/gemini-imagen-model-name-validation

ikaadil committed:
fix: enable tool choice support for model prices and context window

- Updated the "supports_tool_choice" field to true in the model_prices_and_context_window.json file, allowing for tool choice functionality in the specified model.

Signed-off-by: Ifta Khairul Alam Adil <ikaadil007@gmail.com>

ikaadil committed:
Merge branch 'main' into fix/gemini-imagen-model-name-validation
